### PR TITLE
Add Run/Debug Function App actions into Project context menu if applicable

### PR DESCRIPTION
**Pull request by maartenba**
Related to #382, there was a bug related to file paths that caused this feature to stop working (no matching paths == no menu entry). Using FS independent paths everywhere now.

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt [hunk 1]
```diff
@@ -45,13 +45,14 @@ class AzureFunctionsHostRunConfigurationProducer
             configuration: AzureFunctionsHostConfiguration,
             context: ConfigurationContext
     ) : Boolean {
-        val selectedProjectFilePath = context.getSelectedProject()?.getFile()?.path  ?: return false
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt:48]: `File` has an extension `systemIndependentPath`. Can be replaced with:
```
val selectedProjectFilePath = context.getSelectedProject()?.getFile()?.systemIndependentPath  ?: return false
```

**Review by sdubov — APPROVED** [review]
LGTM. One small comment.

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt [hunk 2]
```diff
@@ -45,13 +45,14 @@ class AzureFunctionsHostRunConfigurationProducer
...
+        val selectedProjectFilePath = context.getSelectedProject()?.getFile()?.path ?: return false
```
Comment (maartenba) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt:48]:
```suggestion
        val selectedProjectFilePath = context.getSelectedProject()?.getFile()?.systemIndependentPath  ?: return false
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt [hunk 3]
```diff
@@ -45,13 +45,14 @@ class AzureFunctionsHostRunConfigurationProducer
...
+        val selectedProjectFilePathInvariant = FileUtil.toSystemIndependentName(selectedProjectFilePath)
```
Comment (maartenba) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt:49]:
```suggestion
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt [hunk 4]
```diff
@@ -45,13 +45,14 @@ class AzureFunctionsHostRunConfigurationProducer
...
 
         val projects = context.project.solution.runnableProjectsModel.projects.valueOrNull ?: return false
         val runnableProject = projects.firstOrNull {
             it.kind == RunnableProjectKind.AzureFunctions &&
-            it.projectFilePath == selectedProjectFilePath &&
-            configuration.parameters.projectFilePath == selectedProjectFilePath
+            FileUtil.toSystemIndependentName(it.projectFilePath) == selectedProjectFilePathInvariant &&
+            FileUtil.toSystemIndependentName(configuration.parameters.projectFilePath) == selectedProjectFilePathInvariant
```
Comment (maartenba) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt:55]:
```suggestion
            FileUtil.toSystemIndependentName(it.projectFilePath) == selectedProjectFilePath &&
            FileUtil.toSystemIndependentName(configuration.parameters.projectFilePath) == selectedProjectFilePath
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt [hunk 5]
```diff
@@ -63,13 +64,17 @@ class AzureFunctionsHostRunConfigurationProducer
             ref: Ref<PsiElement>
     ): Boolean {
         val selectedProjectFilePath = context.getSelectedProject()?.getFile()?.path  ?: return false
+        val selectedProjectFilePathInvariant = FileUtil.toSystemIndependentName(selectedProjectFilePath)
```
Comment (maartenba) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt]:
```suggestion
        val selectedProjectFilePathInvariant = context.getSelectedProject()?.getFile()?.systemIndependentPath ?: return false
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt (hunk 3 above)
Comment (maartenba) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostRunConfigurationProducer.kt]:
```suggestion
        val selectedProjectFilePathInvariant = context.getSelectedProject()?.getFile()?.systemIndependentPath ?: return false
```

**Pull request merged**